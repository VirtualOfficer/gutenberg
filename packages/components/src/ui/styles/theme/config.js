/**
 * Internal dependencies
 */
import { get } from '../../create-styles';
import { rgba } from '../../../utils/colors';

const ANIMATION_PROPS = {
	transitionDuration: '200ms',
	transitionDurationFast: '160ms',
	transitionDurationFaster: '120ms',
	transitionDurationFastest: '100ms',
	transitionTimingFunction: 'cubic-bezier(0.08, 0.52, 0.52, 1)',
	transitionTimingFunctionControl: 'cubic-bezier(0.12, 0.8, 0.32, 1)',
};

const GRAY_COLORS = {
	black: '#000',
	gray900: '#1e1e1e',
	gray700: '#757575',
	gray600: '#949494',
	gray500: '#bbb',
	gray400: '#ccc',
	gray300: '#ddd',
	gray200: '#e0e0e0',
	gray100: '#f0f0f0',
	white: '#fff',
};

const MISC_COLORS = {
	darkThemeFocus: get( 'white' ),
	darkGrayPlaceholder: rgba( GRAY_COLORS.gray900, 0.62 ),
	mediumGrayPlaceholder: rgba( GRAY_COLORS.gray900, 0.55 ),
	lightGrayPlaceholdeR: rgba( GRAY_COLORS.white, 0.65 ),
};

const ALERT_COLORS = {
	alertYellow: '#f0b849',
	alertRed: '#cc1818',
	alertGreen: '#4ab866',
};

const COLOR_PROPS = {
	...GRAY_COLORS,
	...MISC_COLORS,
	...ALERT_COLORS,
	colorAdmin: '#007cba',
	colorDestructive: '#D94F4F',
	colorBodyBackground: get( 'white' ),
	colorText: get( 'gray900' ),
	colorTextInverted: get( 'white' ),
	colorTextHeading: '#050505',
	colorTextMuted: '#717171',
};

const FONT_PROPS = {
	fontFamily:
		'-apple-system, BlinkMacSystemFont,"Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell,"Helvetica Neue", sans-serif',
	fontFamilyMono: 'Menlo, Consolas, monaco, monospace',
	fontSize: '13px',
	fontSizeH1: `calc(2.44 * ${ get( 'fontSize' ) })`,
	fontSizeH2: `calc(1.95 * ${ get( 'fontSize' ) })`,
	fontSizeH3: `calc(1.56 * ${ get( 'fontSize' ) })`,
	fontSizeH4: `calc(1.25 * ${ get( 'fontSize' ) })`,
	fontSizeH5: `calc(1 * ${ get( 'fontSize' ) })`,
	fontSizeH6: `calc(0.8 * ${ get( 'fontSize' ) })`,
	fontSizeInputMobile: '16px',
	fontSizeMobile: '15px',
	fontSizeSmall: `calc(0.92 * ${ get( 'fontSize' ) })`,
	fontSizeXSmall: `calc(0.75 * ${ get( 'fontSize' ) })`,
	fontLineHeightBase: '1.2',
	fontWeight: 'normal',
	fontWeightHeading: '600',
};

const SPACING_PROPS = {
	gridBase: '4px',
};

export const config = {
	...COLOR_PROPS,
	// Base
	...FONT_PROPS,
	...SPACING_PROPS,
	// Animations
	...ANIMATION_PROPS,
};
