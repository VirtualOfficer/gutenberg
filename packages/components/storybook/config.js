/**
 * External dependencies
 */
import { addDecorator, configure } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';
import { withInfo } from '@storybook/addon-info';

/**
 * Internal dependencies
 */
import '../build-style/style.css';

// global info settings
addDecorator( withInfo( {
	inline: true, // show source inline, not clicking a button
} ) );

addDecorator( withA11y );
configure(
	[
		require.context( '../docs', true, /\/.+\.mdx$/ ),
		require.context( '../src', true, /\/stories\/.+\.js$/ ),
	],
	module
);
