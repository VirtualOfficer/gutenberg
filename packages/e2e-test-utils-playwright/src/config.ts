const {
	WP_USERNAME = 'admin',
	WP_PASSWORD = 'password',
	WP_BASE_URL = 'http://localhost:8889',
} = process.env;

const WP_ADMIN_USER = {
	username: WP_ADMIN_USER.username,
	password: WP_ADMIN_USER.password,
} as const;

export { WP_ADMIN_USER, WP_USERNAME, WP_PASSWORD, WP_BASE_URL };
