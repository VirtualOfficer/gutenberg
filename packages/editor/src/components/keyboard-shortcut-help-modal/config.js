/**
 * WordPress dependencies
 */
import { displayShortcut } from '@wordpress/keycodes';
import { __ } from '@wordpress/i18n';

const globalShortcuts = {
	title: __( 'Global shortcuts' ),
	description: __( 'Use these shortcuts anywhere.' ),
	shortcuts: [
		{
			key: displayShortcut.primary( '?' ),
			description: __( 'Display this help.' ),
		},
		{
			key: displayShortcut.primary( 's' ),
			description: __( 'Save your changes.' ),
		},
		{
			key: displayShortcut.primary( ',' ),
			description: __( 'Show or hide the settings sidebar.' ),
		},
		{
			key: displayShortcut.primary( 'z' ),
			description: __( 'Undo your last changes.' ),
		},
		{
			key: displayShortcut.primaryShift( 'z' ),
			description: __( 'Redo your last undo.' ),
		},
		{
			key: displayShortcut.secondary( 'm' ),
			description: __( 'Switch between Visual Editor and Code Editor.' ),
		},
	],
};

const selectionShortcuts = {
	title: __( 'Selection shortcuts' ),
	description: __( 'Shortcuts for selecting either text or blocks.' ),
	shortcuts: [
		{
			key: displayShortcut.primary( 'a' ),
			description: __( 'Select all.' ),
		},
		{
			key: 'escape',
			description: __( 'Clear selection.' ),
		},
	],
};

const blockShortcuts = {
	title: __( 'Block shortcuts' ),
	description: __( 'Common shortcuts for use while editing blocks.' ),
	shortcuts: [
		{
			key: displayShortcut.primary( 'del' ),
			description: __( 'Delete the select block(s).' ),
		},
		{
			key: displayShortcut.primary( 'd' ),
			description: __( 'Duplicate the select block(s).' ),
		},
	],
};

const slashInserterShortcuts = {
	title: __( 'Slash inserter' ),
	description: __( 'When adding a new block using the keyboard, use the slash inserter to quickly change the type of block.' ),
	shortcuts: [
		{
			key: '/',
			description: __( 'Search for and change the type of block.' ),
		},
	],
};

const textFormattingShortcuts = {
	title: __( 'Text formatting' ),
	description: __( 'Change the appearance of text.' ),
	shortcuts: [
		{
			key: displayShortcut.primary( 'b' ),
			description: __( 'Make the selected text bold.' ),
		},
		{
			key: displayShortcut.primary( 'i' ),
			description: __( 'Make the selected text italic.' ),
		},
		{
			key: displayShortcut.access( 'd' ),
			description: __( 'Add a strikethrough to the selected text.' ),
		},
		{
			key: displayShortcut.primary( 'k' ),
			description: __( 'Convert the selected text into a link' ),
		},
	],
};

export {
	globalShortcuts,
	selectionShortcuts,
	blockShortcuts,
	slashInserterShortcuts,
	textFormattingShortcuts,
};
