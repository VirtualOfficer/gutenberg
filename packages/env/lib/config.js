'use strict';
/**
 * External dependencies
 */
const fs = require( 'fs' ).promises;
const path = require( 'path' );
const os = require( 'os' );
const crypto = require( 'crypto' );

/**
 * Internal dependencies
 */
const detectDirectoryType = require( './detect-directory-type' );

/**
 * Error subtype which indicates that an expected validation erorr occured
 * while reading wp-env configuration.
 */
class ValidationError extends Error {}

/**
 * The string at the beginning of a source path that points to a home-relative
 * directory. Will be '~/' on unix environments and '~\' on Windows.
 */
const HOME_PATH_PREFIX = `~${ path.sep }`;

/**
 * wp-env configuration.
 *
 * @typedef WPConfig
 * @property {string}      name                     Name of the environment.
 * @property {string}      configDirectoryPath      Path to the .wp-env.json file.
 * @property {string}      workDirectoryPath        Path to the work directory located in ~/.wp-env.
 * @property {string}      dockerComposeConfigPath  Path to the docker-compose.yml file.
 * @property {Object.<string, WPServiceConfig>} env Specific config for different services.
 * @property {boolean}     debug                    True if debug mode is enabled.
 */

/**
 * Base-level config for any particular service. (development/tests/etc)
 *
 * @typedef WPServiceConfig
 * @property {?WPSource}                 coreSource    The WordPress installation to load in the environment.
 * @property {WPSource[]}                pluginSources Plugins to load in the environment.
 * @property {WPSource[]}                themeSources  Themes to load in the environment.
 * @property {number}                    port          The port on which to start the development WordPress environment.
 * @property {Object}                    config        Mapping of wp-config.php constants to their desired values.
 * @property {Object.<string, WPSource>} mappings      Mapping of WordPress directories to local directories which should be mounted.
 */

/**
 * A WordPress installation, plugin or theme to be loaded into the environment.
 *
 * @typedef WPSource
 * @property {'local'|'git'|'zip'} type     The source type.
 * @property {string}              path     The path to the WordPress installation, plugin or theme.
 * @property {?string}             url      The URL to the source download if the source type is not local.
 * @property {?string}             ref      The git ref for the source if the source type is 'git'.
 * @property {string}              basename Name that identifies the WordPress installation, plugin or theme.
 */

module.exports = {
	ValidationError,

	/**
	 * Reads and parses the given .wp-env.json file into a wp-env config object.
	 *
	 * @param {string} configPath Path to the .wp-env.json file.
	 *
	 * @return {WPConfig} A wp-env config object.
	 */
	async readConfig( configPath ) {
		const configDirectoryPath = path.dirname( configPath );
		const workDirectoryPath = path.resolve(
			getHomeDirectory(),
			md5( configPath )
		);

		// The specified base configuration from .wp-env.json or from the local
		// source type which was automatically detected.
		const baseConfig =
			( await getConfigFileData( '.wp-env.json', configPath ) ) ||
			getDefaultBaseConfig();

		// Overriden .wp-env.json on a per-user case.
		const overrideConfig =
			( await getConfigFileData(
				'.wp-env.override.json',
				configPath.replace( /\.wp-env\.json$/, '.wp-env.override.json' )
			) ) || {};

		// A handy merge function which lets us deep-merge the wp-config values
		// instead of overwriting them. This is for merging environment
		const mergeWpServiceConfigs = ( ...configs ) => {
			// Returns an array of nested values in the config object. For example,
			// an array of all the wp-config objects.
			const getNestedValues = ( key, defaultValue = {} ) =>
				configs.map( ( config ) => config[ key ] || defaultValue );

			return {
				...Object.assign( {}, ...configs ),
				config: {
					...Object.assign( {}, ...getNestedValues( 'config' ) ),
				},
			};
		};

		// Configuration applicable to all environments.
		const defaultWpServiceConfig = mergeWpServiceConfigs(
			{
				core: null,
				plugins: [],
				themes: [],
				port: 8888,
				mappings: {},
				config: {
					WP_DEBUG: true,
					SCRIPT_DEBUG: true,
					WP_PHP_BINARY: 'php',
					WP_TESTS_EMAIL: 'admin@example.org',
					WP_TESTS_TITLE: 'Test Blog',
					WP_TESTS_DOMAIN: 'http://localhost',
					WP_SITEURL: 'http://localhost',
					WP_HOME: 'http://localhost',
				},
			},
			baseConfig,
			overrideConfig
		);

		// Configuration specific to each environment. Will be merged with any
		// other specified environments.
		const environmentDefaults = {
			env: {
				development: {}, // No overrides needed, but it should exist.
				tests: {
					config: { WP_DEBUG: false, SCRIPT_DEBUG: false },
					port: 8889,
				},
			},
		};

		// A unique array of the environments specified in the config options.
		// Needed so that we can override settings per-environment, rather than
		// overwriting each environment key.
		const getEnvKeys = ( config ) => Object.keys( config.env || {} );
		const allEnvs = [
			...new Set( [
				...getEnvKeys( environmentDefaults ),
				...getEnvKeys( baseConfig ),
				...getEnvKeys( overrideConfig ),
			] ),
		];

		const getEnvConfig = ( config, envName ) =>
			config.env && config.env[ envName ] ? config.env[ envName ] : {};

		// Merge each of the specified environment-level overrides.
		const env = allEnvs.reduce( ( result, environment ) => {
			result[ environment ] = validateConfig(
				mergeWpServiceConfigs(
					defaultWpServiceConfig,
					getEnvConfig( environmentDefaults, environment ),
					getEnvConfig( baseConfig, environment ),
					getEnvConfig( overrideConfig, environment )
				),
				{
					workDirectoryPath,
					environment,
				}
			);
			return result;
		}, {} );

		return withOverrides( {
			name: path.basename( configDirectoryPath ),
			dockerComposeConfigPath: path.resolve(
				workDirectoryPath,
				'docker-compose.yml'
			),
			configDirectoryPath,
			workDirectoryPath,
			env,
		} );
	},
};

async function getDefaultBaseConfig( configPath ) {
	const configDirectoryPath = path.dirname( configPath );
	const type = await detectDirectoryType( configDirectoryPath );

	if ( type === 'core' ) {
		return { core: '.' };
	} else if ( type === 'plugin' ) {
		return { plugins: [ '.' ] };
	} else if ( type === 'theme' ) {
		return { themes: [ '.' ] };
	}

	throw new ValidationError(
		`No .wp-env.json file found at '${ configPath }' and could not determine if '${ configDirectoryPath }' is a WordPress installation, a plugin, or a theme.`
	);
}

async function getConfigFileData( name, configPath ) {
	try {
		return withBackCompat(
			JSON.parse( await fs.readFile( configPath, 'utf8' ) )
		);
	} catch ( error ) {
		if ( error.code === 'ENOENT' ) {
			return null;
		} else if ( error instanceof SyntaxError ) {
			throw new ValidationError(
				`Invalid ${ name }: ${ error.message }`
			);
		} else {
			throw new ValidationError(
				`Could not read ${ name }: ${ error.message }`
			);
		}
	}
}

/**
 * Used to maintain back compatibility with older versions of the .wp-env.json
 * file. Returns an object in the shape of the currently expected .wp-env.json
 * version.
 *
 * @param {Object} rawConfig config right after being read from a file.
 */
function withBackCompat( rawConfig ) {
	// Convert testsPort into new env.tests format.
	if ( rawConfig.testsPort ) {
		rawConfig.env = {
			...( rawConfig.env || {} ),
			tests: {
				...( rawConfig.env && rawConfig.env.tests
					? rawConfig.env.tests
					: {} ),
				port: rawConfig.testsPort,
			},
		};
	}
	return rawConfig;
}

/**
 * Overrides keys in the config object with set environment variables or options
 * which should be merged.
 *
 * @param {WPConfig} config fully parsed configuration object.
 * @return {WPConfig} configuration object with overrides applied.
 */
function withOverrides( config ) {
	// Override port numbers with environment variables.
	config.env.development.port =
		getNumberFromEnvVariable( 'WP_ENV_PORT' ) ||
		config.env.development.port;
	config.env.tests.port =
		getNumberFromEnvVariable( 'WP_ENV_TESTS_PORT' ) ||
		config.env.tests.port;

	const updateEnvUrl = (
		configKey,
		environments = [ 'development', 'tests' ]
	) => {
		environments.forEach( ( envKey ) => {
			try {
				const baseUrl = new URL(
					config.env[ envKey ].config[ configKey ]
				);
				baseUrl.port = config.env[ envKey ].port;
				config.env[ envKey ].config[ configKey ] = baseUrl.toString();
			} catch ( error ) {
				throw new ValidationError(
					`Invalid .wp-env.json: config.${ configKey } must be a valid URL.`
				);
			}
		} );
	};

	// Update wp config options to include the correct port number in the URL.
	updateEnvUrl( 'WP_TESTS_DOMAIN' );
	updateEnvUrl( 'WP_SITEURL' );
	updateEnvUrl( 'WP_HOME' );

	return config;
}

/**
 * Parses and validates a config object.
 *
 * @param {Object} config A raw config object to parse
 * @param {Object} options
 * @param {string} options.workDirectoryPath Path to the work directory located in ~/.wp-env.
 * @param {string} options.environment Environment name for the service we are parsing.
 * @return {WPServiceConfig} validated and parsed service-level configuration.
 */
function validateConfig( config, options ) {
	if ( config.core !== null && typeof config.core !== 'string' ) {
		throw new ValidationError(
			'Invalid .wp-env.json: "core" must be null or a string.'
		);
	}

	if (
		! Array.isArray( config.plugins ) ||
		config.plugins.some( ( plugin ) => typeof plugin !== 'string' )
	) {
		throw new ValidationError(
			'Invalid .wp-env.json: "plugins" must be an array of strings.'
		);
	}

	if (
		! Array.isArray( config.themes ) ||
		config.themes.some( ( theme ) => typeof theme !== 'string' )
	) {
		throw new ValidationError(
			'Invalid .wp-env.json: "themes" must be an array of strings.'
		);
	}

	if ( ! Number.isInteger( config.port ) ) {
		throw new ValidationError(
			'Invalid .wp-env.json: "port" must be an integer.'
		);
	}

	if ( typeof config.config !== 'object' ) {
		throw new ValidationError(
			'Invalid .wp-env.json: "config" must be an object.'
		);
	}

	if ( typeof config.mappings !== 'object' ) {
		throw new ValidationError(
			'Invalid .wp-env.json: "mappings" must be an object.'
		);
	}

	for ( const [ wpDir, localDir ] of Object.entries( config.mappings ) ) {
		if ( ! localDir || typeof localDir !== 'string' ) {
			throw new ValidationError(
				`Invalid .wp-env.json: "mapping.${ wpDir }" should be a string.`
			);
		}
	}

	return {
		port: config.port,
		coreSource: includeTestsPath(
			parseSourceString( config.core, options ),
			options
		),
		pluginSources: config.plugins.map( ( sourceString ) =>
			parseSourceString( sourceString, options )
		),
		themeSources: config.themes.map( ( sourceString ) =>
			parseSourceString( sourceString, options )
		),
		config: config.config,
		mappings: Object.entries( config.mappings ).reduce(
			( result, [ wpDir, localDir ] ) => {
				const source = parseSourceString( localDir, options );
				result[ wpDir ] = source;
				return result;
			},
			{}
		),
	};
}

/**
 * Parses a source string into a source object.
 *
 * @param {?string} sourceString The source string. See README.md for documentation on valid source string patterns.
 * @param {Object} options
 * @param {string} options.workDirectoryPath Path to the work directory located in ~/.wp-env.
 *
 * @return {?WPSource} A source object.
 */
function parseSourceString( sourceString, { workDirectoryPath } ) {
	if ( sourceString === null ) {
		return null;
	}

	if (
		sourceString.startsWith( '.' ) ||
		sourceString.startsWith( HOME_PATH_PREFIX ) ||
		path.isAbsolute( sourceString )
	) {
		let sourcePath;
		if ( sourceString.startsWith( HOME_PATH_PREFIX ) ) {
			sourcePath = path.resolve(
				os.homedir(),
				sourceString.slice( HOME_PATH_PREFIX.length )
			);
		} else {
			sourcePath = path.resolve( sourceString );
		}
		const basename = path.basename( sourcePath );
		return {
			type: 'local',
			path: sourcePath,
			basename,
		};
	}

	const zipFields = sourceString.match(
		/^https?:\/\/([^\s$.?#].[^\s]*)\.zip$/
	);

	if ( zipFields ) {
		const wpOrgFields = sourceString.match(
			/^https?:\/\/downloads\.wordpress\.org\/(?:plugin|theme)\/([^\s\.]*)([^\s]*)?\.zip$/
		);
		const basename = wpOrgFields
			? encodeURIComponent( wpOrgFields[ 1 ] )
			: encodeURIComponent( zipFields[ 1 ] );
		return {
			type: 'zip',
			url: sourceString,
			path: path.resolve( workDirectoryPath, basename ),
			basename,
		};
	}

	const gitHubFields = sourceString.match( /^([^\/]+)\/([^#]+)(?:#(.+))?$/ );
	if ( gitHubFields ) {
		return {
			type: 'git',
			url: `https://github.com/${ gitHubFields[ 1 ] }/${ gitHubFields[ 2 ] }.git`,
			ref: gitHubFields[ 3 ] || 'master',
			path: path.resolve( workDirectoryPath, gitHubFields[ 2 ] ),
			basename: gitHubFields[ 2 ],
		};
	}

	throw new ValidationError(
		`Invalid or unrecognized source: "${ sourceString }."`
	);
}

/**
 * Given a source object, returns a new source object with the testsPath
 * property set correctly. Only the 'core' source requires a testsPath.
 *
 * @param {?WPSource} source                    A source object.
 * @param {Object}  options
 * @param {string}  options.workDirectoryPath Path to the work directory located in ~/.wp-env.
 *
 * @return {?WPSource} A source object.
 */
function includeTestsPath( source, { workDirectoryPath } ) {
	if ( source === null ) {
		return null;
	}

	return {
		...source,
		testsPath: path.resolve(
			workDirectoryPath,
			'tests-' + path.basename( source.path )
		),
	};
}

/**
 * Parses an environment variable which should be a number.
 *
 * Throws an error if the variable cannot be parsed to a number.
 * Returns null if the environment variable has not been specified.
 *
 * @param {string} varName The environment variable to check (e.g. WP_ENV_PORT).
 * @return {null|number} The number. Null if it does not exist.
 */
function getNumberFromEnvVariable( varName ) {
	// Allow use of the default if it does not exist.
	if ( ! process.env[ varName ] ) {
		return null;
	}

	const maybeNumber = parseInt( process.env[ varName ] );

	// Throw an error if it is not parseable as a number.
	if ( isNaN( maybeNumber ) ) {
		throw new ValidationError(
			`Invalid environment variable: ${ varName } must be a number.`
		);
	}

	return maybeNumber;
}

/**
 * Gets the `wp-env` home directory in which generated files are created.
 *
 * By default, '~/.wp-env/'. On Linux, '~/wp-env/'. Can be overriden with the
 * WP_ENV_HOME environment variable.
 *
 * @return {Promise<string>} The absolute path to the `wp-env` home directory.
 */
async function getHomeDirectory() {
	// Allow user to override download location.
	if ( process.env.WP_ENV_HOME ) {
		return path.resolve( process.env.WP_ENV_HOME );
	}

	/**
	 * Installing docker with Snap Packages on Linux is common, but does not
	 * support hidden directories. Therefore we use a public directory on Linux.
	 *
	 * @see https://github.com/WordPress/gutenberg/issues/20180#issuecomment-587046325
	 */
	return path.resolve(
		os.homedir(),
		!! ( await fs.stat( '/snap' ).catch( () => false ) )
			? 'wp-env'
			: '.wp-env'
	);
}

/**
 * Hashes the given string using the MD5 algorithm.
 *
 * @param {string} data The string to hash.
 * @return {string} An MD5 hash string.
 */
function md5( data ) {
	return crypto.createHash( 'md5' ).update( data ).digest( 'hex' );
}
