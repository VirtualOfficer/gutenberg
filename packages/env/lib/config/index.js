/**
 * Internal dependencies
 */
const { ValidationError, readConfig } = require( './config' );

/**
 * @typedef {import('./config').WPConfig} WPConfig
 * @typedef {import('./config').WPServiceConfig} WPServiceConfig
 * @typedef {import('./config').WPSource} WPSource
 */

module.exports = {
	ValidationError,
	readConfig,
};
