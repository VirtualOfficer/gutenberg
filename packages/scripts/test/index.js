test( 'should leave webpack.config.js untouched', () => {
	expect(
		require( '@wordpress/scripts/config/webpack.config' )
	).toMatchSnapshot();
} );
