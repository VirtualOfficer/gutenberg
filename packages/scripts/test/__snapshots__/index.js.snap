// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should leave webpack.config.js untouched 1`] = `
{
  "devServer": {
    "allowedHosts": "auto",
    "devMiddleware": {
      "writeToDisk": true,
    },
    "host": "localhost",
    "port": 8887,
    "proxy": {
      "/build": {
        "pathRewrite": {
          "^/build": "",
        },
      },
    },
  },
  "devtool": "source-map",
  "entry": [Function],
  "experiments": {
    "outputModule": true,
  },
  "mode": "development",
  "module": {
    "rules": [
      {
        "enforce": "pre",
        "exclude": [
          /node_modules/,
        ],
        "test": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
        "use": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/source-map-loader/dist/cjs.js",
      },
      {
        "exclude": /node_modules/,
        "test": /\\\\\\.m\\?\\(j\\|t\\)sx\\?\\$/,
        "use": [
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/babel-loader/lib/index.js",
            "options": {
              "cacheDirectory": true,
            },
          },
        ],
      },
      {
        "test": /\\\\\\.css\\$/,
        "use": [
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/css-loader/dist/cjs.js",
            "options": {
              "importLoaders": 1,
              "modules": {
                "auto": true,
              },
              "sourceMap": true,
            },
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/postcss-loader/dist/cjs.js",
            "options": {
              "postcssOptions": {
                "ident": "postcss",
                "plugins": [
                  {
                    "browsers": undefined,
                    "info": [Function],
                    "options": {
                      "grid": true,
                    },
                    "postcssPlugin": "autoprefixer",
                    "prepare": [Function],
                  },
                ],
                "sourceMap": true,
              },
            },
          },
        ],
      },
      {
        "test": /\\\\\\.pcss\\$/,
        "use": [
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/css-loader/dist/cjs.js",
            "options": {
              "importLoaders": 1,
              "modules": {
                "auto": true,
              },
              "sourceMap": true,
            },
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/postcss-loader/dist/cjs.js",
            "options": {
              "postcssOptions": {
                "ident": "postcss",
                "plugins": [
                  {
                    "browsers": undefined,
                    "info": [Function],
                    "options": {
                      "grid": true,
                    },
                    "postcssPlugin": "autoprefixer",
                    "prepare": [Function],
                  },
                ],
                "sourceMap": true,
              },
            },
          },
        ],
      },
      {
        "test": /\\\\\\.\\(sc\\|sa\\)ss\\$/,
        "use": [
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/mini-css-extract-plugin/dist/loader.js",
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/css-loader/dist/cjs.js",
            "options": {
              "importLoaders": 1,
              "modules": {
                "auto": true,
              },
              "sourceMap": true,
            },
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/postcss-loader/dist/cjs.js",
            "options": {
              "postcssOptions": {
                "ident": "postcss",
                "plugins": [
                  {
                    "browsers": undefined,
                    "info": [Function],
                    "options": {
                      "grid": true,
                    },
                    "postcssPlugin": "autoprefixer",
                    "prepare": [Function],
                  },
                ],
                "sourceMap": true,
              },
            },
          },
          {
            "loader": "/Users/jonsurrell/a8c/gutenberg/trunk/node_modules/sass-loader/dist/cjs.js",
            "options": {
              "sourceMap": true,
            },
          },
        ],
      },
      {
        "issuer": /\\\\\\.\\(j\\|t\\)sx\\?\\$/,
        "test": /\\\\\\.svg\\$/,
        "type": "javascript/auto",
        "use": [
          "@svgr/webpack",
          "url-loader",
        ],
      },
      {
        "issuer": /\\\\\\.\\(pc\\|sc\\|sa\\|c\\)ss\\$/,
        "test": /\\\\\\.svg\\$/,
        "type": "asset/inline",
      },
      {
        "generator": {
          "filename": "images/[name].[hash:8][ext]",
        },
        "test": /\\\\\\.\\(bmp\\|png\\|jpe\\?g\\|gif\\|webp\\)\\$/i,
        "type": "asset/resource",
      },
      {
        "generator": {
          "filename": "fonts/[name].[hash:8][ext]",
        },
        "test": /\\\\\\.\\(woff\\|woff2\\|eot\\|ttf\\|otf\\)\\$/i,
        "type": "asset/resource",
      },
    ],
  },
  "optimization": {
    "concatenateModules": false,
    "minimizer": [
      TerserPlugin {
        "options": {
          "exclude": undefined,
          "extractComments": false,
          "include": undefined,
          "minimizer": {
            "implementation": [Function],
            "options": {
              "compress": {
                "passes": 2,
              },
              "mangle": {
                "reserved": [
                  "__",
                  "_n",
                  "_nx",
                  "_x",
                ],
              },
              "output": {
                "comments": /translators:/i,
              },
            },
          },
          "parallel": true,
          "test": /\\\\\\.\\[cm\\]\\?js\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        },
      },
    ],
    "splitChunks": {
      "cacheGroups": {
        "default": false,
        "style": {
          "chunks": "all",
          "enforce": true,
          "name": [Function],
          "test": /\\[\\\\\\\\/\\]style\\(\\\\\\.module\\)\\?\\\\\\.\\(pc\\|sc\\|sa\\|c\\)ss\\$/,
          "type": "css/mini-extract",
        },
      },
    },
  },
  "output": {
    "filename": "[name].js",
    "path": "/Users/jonsurrell/a8c/gutenberg/trunk/build",
  },
  "plugins": [
    DefinePlugin {
      "definitions": {
        "SCRIPT_DEBUG": true,
      },
    },
    CleanWebpackPlugin {
      "apply": [Function],
      "cleanAfterEveryBuildPatterns": [
        "!fonts/**",
        "!images/**",
      ],
      "cleanOnceBeforeBuildPatterns": [
        "**/*",
      ],
      "cleanStaleWebpackAssets": false,
      "currentAssets": [],
      "dangerouslyAllowCleanPatternsOutsideProject": false,
      "dry": false,
      "handleDone": [Function],
      "handleInitial": [Function],
      "initialClean": false,
      "outputPath": "",
      "protectWebpackAssets": true,
      "removeFiles": [Function],
      "verbose": false,
    },
    RenderPathsPlugin {},
    CopyPlugin {
      "options": {},
      "patterns": [
        {
          "context": "src",
          "from": "**/block.json",
          "noErrorOnMissing": true,
          "transform": [Function],
        },
        {
          "context": "src",
          "filter": [Function],
          "from": "**/*.php",
          "noErrorOnMissing": true,
        },
      ],
    },
    MiniCssExtractPlugin {
      "_sortedModulesCache": WeakMap {},
      "options": {
        "chunkFilename": "[name].css",
        "experimentalUseImportModule": undefined,
        "filename": "[name].css",
        "ignoreOrder": false,
        "runtime": true,
      },
      "runtimeOptions": {
        "attributes": undefined,
        "insert": undefined,
        "linkType": "text/css",
      },
    },
    DependencyExtractionWebpackPlugin {
      "externalizedDeps": Set {},
      "options": {
        "combineAssets": false,
        "combinedOutputFile": null,
        "externalizedReport": false,
        "injectPolyfill": false,
        "outputFilename": null,
        "outputFormat": "php",
        "useDefaults": true,
      },
      "useModules": false,
    },
  ],
  "resolve": {
    "alias": {
      "lodash-es": "lodash",
    },
    "extensions": [
      ".jsx",
      ".ts",
      ".tsx",
      "...",
    ],
  },
  "stats": {
    "children": false,
  },
  "target": "browserslist",
}
`;
